### ISSUE TITLE
Not much faster for me

### ISSUE THREAD

**Issue by vbraun** [cspotcode/typescript-transpile-only#2]
Not a bug, just a comment / datapoint: In my project (2186 files) the new tsc incremental build option is faster than turning off type checking.

tsc-transpile-only builds in about 18s:
~~~~
$ time tsc-transpile-only -p build.dev/typescript/tsconfig-app.json 

real	0m17.829s
user	0m32.632s
sys	0m0.792s
~~~~

tsc cold build is 21s, incremental is 12s:
~~~~
$ time tsc -p build.dev/typescript/tsconfig-app.json 

real	0m20.863s
user	0m38.351s
sys	0m0.834s
$ time tsc -p build.dev/typescript/tsconfig-app.json 

real	0m11.591s
user	0m22.628s
sys	0m0.507s
~~~~

The quest for faster incremental builds continues ;-)

**Comment by cspotcode** [comment]
Makes sense.  The goal is not speed, but to give a meaningful exit code based on **syntactic** errors, not **semantic** errors.

Incremental builds are able to skip the emit step for files that haven't changed.  This tool cannot, because it doesn't try to read incremental build info.  I suppose it could be updated to use `createIncrementalProgram` which would theoretically pull `tsbuildinfo` from disk automatically.  But as far as I know, those builders have semantic checking baked in, so we couldn't skip it.

We could also integrate this CLI with a caching layer like [typescript-cached-transpile](https://www.npmjs.com/package/typescript-cached-transpile)

EDIT: actually, we don't need to skip semantic checking, just ensure that we still exit with code 0 in that case.  So we could probably use `createIncrementalProgram` to read and write `tsbuildinfo`, getting the speed improvements.
